app/src/main/java/org/fossasia/susi/ai/device/viewdevice/ViewDevicePresenter.kt

```diff
@@ -0,0 +1,82 @@
+package org.fossasia.susi.ai.device.viewdevice
+
+import android.content.Context
+import io.realm.Realm
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.data.UtilModel
+import org.fossasia.susi.ai.data.contract.IRoomModel
+import org.fossasia.susi.ai.data.device.RoomModel
+import org.fossasia.susi.ai.data.model.RoomsAvailable
+import org.fossasia.susi.ai.dataclasses.AddDeviceQuery
+import org.fossasia.susi.ai.device.viewdevice.contract.IViewDevicePresenter
+import org.fossasia.susi.ai.device.viewdevice.contract.IViewDeviceView
+import org.fossasia.susi.ai.rest.responses.susi.Device
+import org.fossasia.susi.ai.rest.responses.susi.GetAddDeviceResponse
+import retrofit2.Response
+import timber.log.Timber
+
+class ViewDevicePresenter : IViewDevicePresenter, IRoomModel.onAddDeviceListener {
+
+    private lateinit var viewDeviceView: IViewDeviceView
+    private var macId: String? = null
+    private var device: Device? = null
+    private lateinit var realm: Realm
+    private val roomModel: RoomModel = RoomModel()
+    private lateinit var context: Context
+    private lateinit var utilModel: UtilModel
+
+    override fun onAttach(viewDeviceView: IViewDeviceView, macId: String?, device: Device?, context: Context) {
+        this.viewDeviceView = viewDeviceView
+        this.macId = macId
+        this.device = device
+        this.context = context
+        realm = Realm.getDefaultInstance()
+        utilModel = UtilModel(context)
+    }
+
+    override fun addRoom(roomName: String) {
+        realm = Realm.getDefaultInstance()
+        realm.beginTransaction()
+        var results = realm.where(RoomsAvailable::class.java).findAll()
+        var id = results.size
+        id++
+
+        var addedRoomModel = realm.createObject(RoomsAvailable::class.java)
+        addedRoomModel.id = id.toLong()
+        addedRoomModel.room = roomName
+        realm.commitTransaction()
+        viewDeviceView.showRooms()
+        // deviceConnectView?.roomNameSelected(null)
```

Comment:
Why?

Reply:
Actually, this statement was used for a feature to deselect previously selected options, when a new room is created. But now, I have removed that feature. So, it was commented. This feature can be re-added just by uncommenting it if required. Hence I kept that code line.